lib/Pool.js

```diff
@@ -437,6 +437,7 @@ class Pool extends EventEmitter {
 
     // TODO: should we defer this check till after this event loop incase "the situation" changes in the meantime
     if (
+      !this.spareResourceCapacity && !this.available &&
```

Comment:
I think I'd prefer `!this.available` to more explicit for easier reading later so something like `this.available < 1`
Also it's preferable to directly reference/query the internal objects/queues/values rather then the external interface so this should be `this._availableObjects.length < 1`.

Whats the intention behind adding this extra check?

_I think the linter is complaining bout the line length or stacking multiple `&&` clauses on this line as well_

Reply:
When we set maxWaitingClients = 0, then the following code will start returning error when we have a acquire session request burst.
```
    // TODO: should we defer this check till after this event loop incase "the situation" changes in the meantime
    if (
      this._config.maxWaitingClients !== undefined &&
      this._waitingClientsQueue.length >= this._config.maxWaitingClients
    ) {
      return this._Promise.reject(
        new Error("max waitingClients count exceeded")
      );
    }
```
ideally, what we want is, we should not throw any error, as long as we are able to create new sessions. When the max capacity is reached and no sessions are available then we should start throwing an error when maxWaitingClients = 0.

Hence the check is added.